apps/src/dance/Dance.js

```diff
@@ -130,6 +131,21 @@ Dance.prototype.init = function (config) {
     config.valueTypeTabShapeMap = {[Blockly.BlockValueType.SPRITE]: 'angle'};
 
     this.studioApp_.init(config);
+
+    if (this.usesPreview) {
+      this.currentCode = this.studioApp_.getCode();
```

Comment:
Minor - do you need to save `this.currentCode` and an instance variable? could this just be a local variable?

Reply:
Yes - can be a local variable (updated now) and I also cleaned up nested if - thanks!

Reply:
Hey @breville  - I found a case when the local variable does not work so I reverted back to using an instance variable.

In preview mode, when there is a block and then it is removed, the preview does not update correctly. The sprite image remains on screen:


https://github.com/code-dot-org/code-dot-org/assets/107423305/9ba3f3dd-81ad-4087-99cd-6c8356a96b99



When `this.currentCode` is used, the preview updates correctly when a block is remove and there are no blocks within `setup`:


https://github.com/code-dot-org/code-dot-org/assets/107423305/152d92dc-d98b-40ab-a76d-6e14c5f4a8c4


Another interesting thing I noticed either way is that a default sprite (gray square) is drawn when the code space is empty, and the user is dragging a block from the toolbox to the workspace.

@breville said that we should probably implement this feature similar to how the timeline works in music lab - that the code is compared when [it is compiled](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/music/blockly/MusicBlocklyWorkspace.js#L208). So if we do, then the issues raised above (using instance variable vs local variable and gray square sprite) may both be addressed?

Reply:
[Update](https://github.com/code-dot-org/code-dot-org/pull/53817/commits/fed4b76561dd41c2208fb90fc89fbaeafb233d19): 
In `StudioApp.js`  I added the event passing to the change handler in `Dance.js`. Similar to [music lab](https://github.com/code-dot-org/code-dot-org/blob/40ae8c3f5a0156cec6c7bef4e8ec302fc7334af8/apps/src/music/views/MusicView.jsx#L329), the event type is checked. However, instead of checking for the 'drag' event, I check to see if the event type is either 'move' or 'change' and if not, then preview is not called. I learned that 'move' is when a block is dragged and the dropped. This resolved the gray sprite appearing briefly when the 'make new dance' block was being dragged from the toolbox to the workspace and the first error @breville described [in this comment](https://github.com/code-dot-org/code-dot-org/pull/53817#issuecomment-1726473182).

Reply:
Screenshot of fixed bug (note that you don't see sprite until the block is dropped within the setup block:

https://github.com/code-dot-org/code-dot-org/assets/107423305/e9ca48b7-d366-4be2-a729-46ac2e8c430c